### PR TITLE
Carousel component

### PR DESCRIPTION
**Pull request by steedems**
## Summary 🍿 

Add carousel component to display slides on different viewports.

## Preview 📸 

See full preview [here](https://deploy-preview-634--zrc.netlify.app/#/Components/Organisms/Carousel)

Desktop - width > 768px
Screenshot 2020-11-10 at 16 37 11

Mobile - width < 768px
Screenshot 2020-11-10 at 16 37 45

## Behind the scenes 🎪 

To prevent the page from jumping up and down when browsing through slides of different height each slide is given a minimum height equal to the height of the highest slide. This is done when the component mounts so if the user resizes the window the height is not computed again and spacing might seem a bit off. I believe this is minor and could be improved in the future if needed.

## TODOs

- [x] unit tests ⛑️

### CODE REVIEW

File: src/components/organisms/Carousel/Carousel/Carousel.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  /**
+   * Used to specify which slide of the carousel should be initially visible.
+   * @default Slide closest to the center: Math.ceil(totalNumberOfSlides / 2)
+   */
+  initialSlide?: number;
+}
+
+const CarouselComponent: FC<CarouselComponentProps> = ({ children, initialSlide }) => {
+  const slidesCount = Children.count(children);
+  const centralSlide = Math.ceil(slidesCount / 2) - 1;
```
Comment (admmasters) [inline src/components/organisms/Carousel/Carousel/Carousel.tsx:21]: maybe centralSlideIndex

File: src/components/organisms/Carousel/Carousel/Carousel.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import FlexContainer from '../../../layout/FlexContainer/FlexContainer';
+import FlexRow from '../../../layout/FlexRow/FlexRow';
+import { CarouselContext } from '../context/CarouselContext';
+
+export interface CarouselComponentProps {
+  /**
+   * Used to specify which slide of the carousel should be initially visible.
+   * @default Slide closest to the center: Math.ceil(totalNumberOfSlides / 2)
+   */
+  initialSlide?: number;
```
Comment (admmasters) [inline src/components/organisms/Carousel/Carousel/Carousel.tsx:16]: /s/initialSlideIndex

**Review by rominamoya — DISMISSED** [review]

File: src/components/organisms/Carousel/Carousel/ Carousel.test.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import { fireEvent, render, screen } from '@testing-library/react';
```
Comment (endymion1818) [inline src/components/organisms/Carousel/Carousel/ Carousel.test.tsx:1]: Is it me or is that an extra character in the filename?

Comment (steedems) [reply src/components/organisms/Carousel/Carousel/ Carousel.test.tsx:1]: lens

**Review by admmasters — APPROVED** [review]

**Pull request merged**